snapshot/controllers.go

```diff
@@ -0,0 +1,92 @@
+package snapshot
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/flanksource/incident-commander/api"
+	"github.com/labstack/echo/v4"
+)
+
+func Topology(c echo.Context) error {
+	id := c.Param("id")
+	related, _ := strconv.ParseBool(c.QueryParam("related"))
+	logStart := c.QueryParam("logStart")
+	logEnd := c.QueryParam("logEnd")
+	if logStart == "" {
+		logStart = "15m"
+	}
+	directory := fmt.Sprintf("snapshot-%s", time.Now().Format(time.RFC3339))
+	if err := os.MkdirAll(directory, os.ModePerm); err != nil {
+		return c.JSON(http.StatusInternalServerError, api.HTTPErrorMessage{
+			Error:   err.Error(),
+			Message: "Error creating directory",
+		})
+	}
+
+	ctx := SnapshotContext{
+		Directory: directory,
+		LogStart:  logStart,
+		LogEnd:    logEnd,
```

Comment:
nit: this entire block could be refactored into NewSanpshotContext(c) which then uses echo.Bind